### PR TITLE
Support cross compilation

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
This is ready for a review

File: recipe/meta.yaml [hunk 1]
```diff
@@ -38,40 +44,36 @@ outputs:
...
         - m2-sed  # [win]
       host:
         # Needed to symlink libcuda into sysroot libs.
-        - {{ compiler("c") }}
-{% if cdt_name == "cos7" %}                    # [linux]
-        - sysroot_{{ target_platform }} 2.17   # [linux]
-        - __glibc >=2.17                       # [linux]
-{% endif %}                                    # [linux]
+        - {{ c_compiler }}_{{ cross_target_platform }} {{ c_compiler_version }}.*       # [linux]
+        - {{ c_compiler }}_{{ cross_target_platform }}                                  # [win]
```
Comment (kkraus14) [inline recipe/meta.yaml:54]: Should we version this for windows in the case that conda-forge wants to generally move to newer versions than what an older version of `nvcc` supports?

Comment (isuruf) [reply recipe/meta.yaml:54]: I think that's unrelated to this PR and should be a separate discussion.

**Review by kkraus14 — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,19 +11,25 @@
...
   name: "{{ name }}"
 
 build:
   number: {{ number }}
 {% if cuda_major == 0
-    or (target_platform == "linux-ppc64le" and cuda_major_minor < (10, 2))
-    or (target_platform == "linux-aarch64" and (cuda_major_minor < (11, 0) or arm_variant_type != "sbsa")) %}
+    or (cross_target_platform == "linux-ppc64le" and cuda_major_minor < (10, 2))
+    or (cross_target_platform == "linux-aarch64" and (cuda_major_minor < (11, 0) or arm_variant_type != "sbsa"))
+    or (cross_target_platform != target_platform and cuda_compiler_version != "11.2") %}
```
Comment (leofang) [inline recipe/meta.yaml:26]: Q: We only support cross-compiling using CUDA 11.2?

**Review by leofang — COMMENTED** [review]
LGTM based on my limited knowledge in cross-compiling support. Thanks, Isuru.

Even with that aside, this PR makes the recipe look nicer by replacing the `cdt_name == "cos7"` jinja conditionals with selectors.

File: recipe/meta.yaml (hunk 2 above)
Comment (isuruf) [reply recipe/meta.yaml:26]: Yes, we added support for 11.2 only in conda-forge-ci-setup-feedstock

**Comment by isuruf** [comment]
Will merge soon, if there are no other comments

**Comment by leofang** [comment]
I missed the ping, but @isuruf feel free to merge anytime.

**Pull request merged**